### DISCUSSION TITLE
Dialogic - 1.4.5

### DISCUSSION THREAD

**Discussion by coppolaemilio** [dialogic-godot/dialogic#1173] [Announcements]
Another small release to fix some bugs in the Godot 3 version of Dialogic

## What's Changed
* Added focus style options for themes [[CptFubar](https://github.com/CptFubar)]
* Fixed a broken font reference [[todd-koeck](https://github.com/todd-koeck)]
* Fix for Nameplate being unaligned [[meyaoigames](https://github.com/meyaoigames)]
* Fix crash if theme is missing on character file [[thebardsrc](https://github.com/thebardsrc)]

## New Contributors
* @CptFubar made their first contribution in https://github.com/coppolaemilio/dialogic/pull/1165

**Full Changelog**: https://github.com/coppolaemilio/dialogic/compare/v1.4.4...1.4.5

<hr /><em>This discussion was created from the release <a href='https://github.com/coppolaemilio/dialogic/releases/tag/1.4.5'>Dialogic - 1.4.5</a>.</em>

**Comment by kyle-grobbelaar** [comment]
I'm trying to use dialogic in my project but whenever I add text to my events, they don't display what text is written just a blank text box. only displaying small portions, such as a few characters each time. Is there a fix for this or am I simply making a mistake somewhere?

**Reply by Jowan-Spooner** [reply]
I assume you mean the text box of the text event in editor, not in game. 
If so, this is a known issue, caused by differences in display size. The best solution is to go into `res://addons/dialogic/Editor/Events/Parts/Text` and open `TextEditor.tscn` select the root node, go into the inspector, scroll down to `Size Flags` and enable horizontal Expand. Save the scene and reload the project.

**Reply by kyle-grobbelaar** [reply]
This worked thank you!

**Comment by Raebaee** [comment]
I tried adding in the dialogue using the step by step tutorial included, but all that happens when I play the game is part of the screen gets covered in an transparent black and the bottom is normal, so no text or portrait is being shown. Is this a normal occurrence or is there just something I'm missing?

**Reply by Jowan-Spooner** [reply]
Hi, this sounds like you have set the window size of your game to be quite small (for example for a pixel art game). Dialogics default theme is designed with the default resolution in mind. To fix this you can create a new theme or edit the default theme. In particular what will be interesting for you is the dialog box size and the text size (which can be edited by clicking the pen icon next to the font and changing the size in the inspector). If you change them (they are in pixels) to something that makes sense with your window size, then it should work better.

**Reply by Raebaee** [reply]
Thank you so much! This helped a lot :)

**Comment by sshearer** [comment]
Hi, I'm a bit behind the times :)  I just updated Godot to v3.5.1-stable and then tried to update Dialogic by:
1. in godot UI, AssetLib, searched for Dialogic, clicked Download, clicked Install, saw the Success pop-up.  But when I went to the Dialogic editor, the top-right corner still showed my previous version (v1.4.2).  Same after exiting & restarting godot.  I checked Project Settings, Add-Ins, it still showed the old version.  ie. i didn't see new version.  So:

2. in godot, Project Settings, Addons, I disabled the current/older version of Dialogic & shutdown godot.  Then in github, I went to your project and downloaded this ZIP file, unzipped, went to the addons folder & renamed the earlier dialogic folder and moved the unzipped dialogic folder into the same addons folder.  Back in godot, Project Settings, Addons, I see both renamed & new 'dialogic' folders, enabled the new v1.4.5 'dialogic' addon.  Saved the project/Scene.  Restarted godot.  When I go to Dialogic 'page', I see:

- "Dialogic v1.4 - DEV' in the top-right corner of the screen, and
- I no longer see ANY of my characters, timelines, or content, and
- the icons in the top-left corner of that screen are all disabled.

So, @coppolaemilio or someone, what did I do wrong, and what do I need to do to complete the upgrade & restore my content ?

**Reply by Jowan-Spooner** [reply]
Best would be to completeley remove the addons/dialogic folder and then re-install. Godots plugin installation sometimes messes up what files to keep/remove and which to replace. As long as you don't touch your res://dialogic folder your timelines and characters should be save.

In the future, use version control ;)

**Comment by Conding-Student** [comment]
Good day, I do have some problem. the dialogic was running when I played in Godot 3.5 engine but when I rendered it on android phone it doesn't starting. It was just a plain background no dialogue, no characters.

I'll make sure to export all resources especially those .json extension in Resources before rendering into APK format also the code to initialize and start the dialogue was neat. My windows size was quite small because it was pixelated game but somehow it was working when I played in PC. 

Do I missed to checked some things?

**Reply by zaknafean** [reply]
Hey, 
So dialogic for Godot 3 has moved to its own repository over at https://github.com/dialogic-godot/dialogic-1
So if you need support asking there will help keep things organized. That said...

Just double check you're exporting the right files:
![image](https://github.com/dialogic-godot/dialogic/assets/7741797/4a8d60bc-76e8-425d-bc13-c0139f8da720)


If that doesn't work and you're actually using Godot 1.4.5, you may be behind enough that you're hitting all the android bugs we fixed for 1.5+. I'd suggest making a backup, and try the latest version from the linked repository. That may clean up your issues.

**Reply by Conding-Student** [reply]
Good day, 
Thank you very much!! it finally works on android phone!! I missed the *.cfg files.  I also used dialogic 1.5. Your my hero Sir, thanks again.
